### ISSUE TITLE
Tibber connectivity

### ISSUE THREAD

**Issue by pittbull** [ottopaulsen/node-red-contrib-power-saver#173]
Yesterday after upgrading HA to 10.1 my capacity live node will not connect to the Tibber api. As per screenshot it will stay in "connecting" mode without giving any info in debug or in NR logfile. 

I have tried to activate/deactivate and disable/enable but it stays the same. 

Do you have any tips?!

[image](https://github.com/ottopaulsen/node-red-contrib-power-saver/assets/1532149/e3a85759-400c-4c57-947f-c6f9fd5b7c28)

**Comment by pittbull** [comment]
Agreed, but I am unable to extract any logs. If I change the API-key I get an error on wrong key, so the link is there - but no errors anywhere...

**Comment by ottopaulsen** [comment]
Did you figure this out?

**Comment by pittbull** [comment]
Well, to be honest, I do not know what solved it. Running out of ideas I disabled and reset a few nodes and suddenly it was all up and running again. I also increased the timeouts on the node at the same time (and reset them to default later). The problem has not reoccured.

**Comment by pittbull** [comment]
Having said that the issue reoccured just this evening after a regular restart.

This is the error:
<img width="319" alt="image" src="https://github.com/ottopaulsen/node-red-contrib-power-saver/assets/1532149/561e5e65-4c72-4cc3-9747-3614bd2b35d7">

And these nodes are not created:
<img width="1055" alt="image" src="https://github.com/ottopaulsen/node-red-contrib-power-saver/assets/1532149/0a16f6a5-02f4-4aa3-a5a7-16147ca8fc35">

And this warning is shown on the "find highest per day"-node:
<img width="709" alt="image" src="https://github.com/ottopaulsen/node-red-contrib-power-saver/assets/1532149/e2dbdcba-d1f4-4378-96b1-51062ae8c3f1">

Could the 400 errors be related to not being able to set the value for these two, as they - for some reason - are unknown?

//UlfThomas

**Comment by pittbull** [comment]
Situation autoresolved itself during the night 🤷‍♂️

**Comment by ottopaulsen** [comment]
It is the Set entity node that reports errors. It is connecting to the Home Assistant API. 
400 means client error, so yes, there may be something wrong in the previous nodes.
That error message was very strange. Looks like a compilation error. The strange thing is that it only happens some times.
Seems to be some bug in the code that occurs only when some data is missing.
Would be interesting to know the inpt to the "find highest per day" node, but I guess that is hard to get hold of now.

**Comment by pittbull** [comment]
Agreed. And yes, for the time being it would be kinda hard to debug. I'll make sure to capture this data if it happens again. 

Thinking about error in previous nodes: could the issue be related to a lack of data (from Tibber?) resulting in code not being able to compute "top 3" in the start of a month? I think I've seen this somewhere else?
